packages/styles/src/components/dialog.scss

```diff
@@ -0,0 +1,301 @@
+    }
+
+    post-dialog-header,
+    post-dialog-body,
+    post-dialog-controls {
+      padding-inline: spacing.$size-regular;
+    }
+
+    post-dialog-close {
+      margin-right: spacing.$size-regular;
+    }
+
+    post-dialog-header,
+    post-dialog-controls {
+      padding-bottom: spacing.$size-regular;
+    }
+  }
+
+  &[data-variant='cookie'] {
+    top: auto;
+    bottom: 2rem;
+
+    > form {
+      grid-template-columns: auto 1fr auto;
+      grid-template-areas:
+        'icon     header   close-button'
+        'icon     body     close-button'
+        'controls controls controls';
+    }
+
+    post-dialog-icon {
+      align-self: flex-start;
+    }
+  }
+}
+
+post-dialog-icon,
+post-dialog-header,
+post-dialog-body,
+post-dialog-controls,
+post-dialog-close {
+  &:empty {
+    display: none;
+  }
+}
+
+post-dialog-icon {
+  grid-area: icon;
+  display: block;
+
+  > post-icon {
+    width: spacing.$size-big;
+    height: spacing.$size-big;
+
+    // Larger icon for bigger notification dialogs
+    dialog:not([size='small']) & {
+      @include media-breakpoint-up(rg) {
+        width: spacing.$size-small-huge;
+        height: spacing.$size-small-huge;
+      }
+    }
+  }
+}
+
+post-dialog-header {
+  grid-area: header;
+
+  & > *:first-child {
+    margin-top: 0;
+  }
+}
+
+post-dialog-body {
+  grid-area: body;
+  margin-bottom: 0;
+
+  & > *:last-child {
+    margin-bottom: 0;
+  }
+}
+
+post-dialog-controls {
+  grid-area: controls;
+  position: sticky;
+  bottom: 0;
+  padding-block: spacing.$size-regular;
+  display: flex;
+  flex-wrap: wrap;
+  flex-direction: row-reverse;
+  gap: spacing.$size-regular;
+  background-color: inherit;
+
+  button {
+    @include media-breakpoint-down(sm) {
+      width: 100%;
+    }
+  }
+}
+
+post-dialog-close {
+  position: sticky;
+  top: 0;
+  grid-area: close-button;
+  width: spacing.$size-large;
+  height: spacing.$size-large;
+  min-height: 0;
+  margin-left: auto;
+}
+
+// Animations
+dialog:not([data-animation='none']) {
+  --_dialog-transition-duration: var(--dialog-transition-duration, 0.5s);
+
+  transform: scale(0.8);
+  opacity: 0;
+  transition-property: transform, opacity, overlay, display;
+  transition-behavior: allow-discrete;
+  transition-duration: var(--_dialog-transition-duration);
+  transition-timing-function: linear(
+    0,
+    0.007,
+    0.029 2.2%,
+    0.118 4.7%,
+    0.625 14.4%,
+    0.826 19%,
+    0.902,
+    0.962,
+    1.008 26.1%,
+    1.041 28.7%,
+    1.064 32.1%,
+    1.07 36%,
+    1.061 40.5%,
+    1.015 53.4%,
+    0.999 61.6%,
+    0.995 71.2%,
+    1
+  );
+
+  &::backdrop {
+    opacity: 0;
+    transition:
+      backdrop-filter var(--_dialog-transition-duration),
+      opacity var(--_dialog-transition-duration),
+      overlay var(--_dialog-transition-duration) allow-discrete,
+      display var(--_dialog-transition-duration) allow-discrete;
+  }
+
+  &[open] {
+    transform: scale(1);
+    opacity: 1;
+
+    @starting-style {
```

Comment:
Both Safari (https://github.com/WebKit/standards-positions/issues/210) and Firefox (https://github.com/mozilla/standards-positions/issues/833) are supportive of the spec, therefore I consider it safe to use for progressive enhancement.